src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Patch.cs

```diff
@@ -0,0 +1,601 @@
+
+                foreach (var typeName in recordType.GetNames(DPath.Root))
+                {
+                    DType type = typeName.Type;
+                    DName columnName = typeName.Name;
+
+                    if (!dsType.Contains(columnName))
+                    {
+                        continue;
+                    }
+
+                    foreach (var tabularDataSource in dsType.AssociatedDataSources)
+                    {
+                        dataSourceToQueryOptionsMap.AddSelect(tabularDataSource, columnName);
+                    }
+                }
+            }
+
+            return true;
+        }
+
+        protected override bool RequiresPagedDataForParamCore(TexlNode[] args, int paramIndex, TexlBinding binding)
+        {
+            Contracts.AssertValue(args);
+            Contracts.AssertAllValues(args);
+            Contracts.Assert(paramIndex >= 0 && paramIndex < args.Length);
+            Contracts.AssertValue(binding);
+            Contracts.Assert(binding.IsPageable(Contracts.VerifyValue(args[paramIndex])));
+
+            // For the first argument, we need only metadata. No actual data from datasource is required.
+            return paramIndex > 0;
+        }
+    }
+    #endregion
+
+    #region Overloads classes
+
+    // Patch(dataSource:*[], Record, Updates1, Updates2,…)
+    internal class PatchFunction : PatchAsyncFunctionCore
+    {
+        public override bool CanSuggestInputColumns => true;
+
+        public override bool TryGetTypeForArgSuggestionAt(int argIndex, out DType type)
+        {
+            if (argIndex > 1)
+            {
+                type = default;
+                return false;
+            }
+
+            return base.TryGetTypeForArgSuggestionAt(argIndex, out type);
+        }
+
+        public PatchFunction()
+            : base("Patch", TexlStrings.AboutPatch, FunctionCategories.Table | FunctionCategories.Behavior, DType.EmptyRecord, 0, 3, int.MaxValue, DType.EmptyTable, DType.EmptyRecord, DType.EmptyRecord)
+        {
+        }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
+        {
+            yield return new[] { TexlStrings.PatchArg_Source, TexlStrings.PatchArg_Record, TexlStrings.PatchArg_Update };
+            yield return new[] { TexlStrings.PatchArg_Source, TexlStrings.PatchArg_Record, TexlStrings.PatchArg_Update, TexlStrings.PatchArg_Update };
+        }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures(int arity)
+        {
+            if (arity > 3)
+            {
+                return GetGenericSignatures(arity, TexlStrings.PatchArg_Source, TexlStrings.PatchArg_Record, TexlStrings.PatchArg_Update);
+            }
+
+            return base.GetSignatures(arity);
+        }
+
+        internal override IntermediateNode CreateIRCallNode(PowerFx.Syntax.CallNode node, IRTranslatorContext context, List<IntermediateNode> args, ScopeSymbol scope)
```

Comment:
>CreateIRCallNode [](http://example.com/codeflow?start=43&length=16)

If this override only calls the base with the same arguments, do we need it?

Reply:
No. Thanks for pointing this out.